### PR TITLE
Fix usage of plain metadata type with new disks configuration option

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix usage of plain metadata type for new disks configuration option.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit d4588f187165635ac045ad79d5451cd2bdbdd80e with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60396/d4588f187165635ac045ad79d5451cd2bdbdd80e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60396/d4588f187165635ac045ad79d5451cd2bdbdd80e/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by vdimir — APPROVED** [review]

File: src/Disks/DiskType.h [hunk 1]
```diff
@@ -17,7 +17,6 @@ enum class ObjectStorageType
 {
     None,
     S3,
-    S3_Plain,
```
Comment (jkartseva) [inline src/Disks/DiskType.h:20]: Are these enum values serialized anywhere?

Comment (kssenii) [reply src/Disks/DiskType.h:20]: In zero copy replication, but it cannot be used with plain metadata type.

**Review by vdimir — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests (tsan) [3/6] — fail: 1

test_storage_azure_blob_storage/test.py::test_union_schema_inference_mode

> Stateless tests (tsan, s3 storage) [1/5] — fail: 1

02784_parallel_replicas_automatic_decision_join

> A Sync — Private CI has failed 

In fact the sync PR is green and all tests passed, the check result update was not triggered.

**Comment by clickgapai** [comment]
Hi @kssenii @vdimir — while reviewing this PR I found the following:

- **Coverage gap:** `local_blob_storage` + `metadata_type=plain` combination has no stateless coverage — https://github.com/ClickHouse/ClickHouse/pull/103889

Could you review the test PR and add the `can be tested` label if it looks good?
Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**